doc/mgr/orchestrator_cli.rst

```diff
@@ -89,23 +89,17 @@ to talk to it)
 
 Also show any in-progress actions.
 
-..
-    Host Management
-    ~ ~~~~~~~~~~~~~~
-
-    List hosts associated with the cluster: :
-
-        ceph orchestrator host ls
+Host Management
+~~~~~~~~~~~~~~~
 
-    Add and remove hosts: :
+List hosts associated with the cluster::
 
-      ceph orchestrator host add <host>
-      ceph orchestrator host rm <host>
+    ceph orchestrator host ls
 
-    . . note: :
-
-    Removing a host only succeeds, if the host is unused.
+Add and remove hosts::
 
+    ceph orchestrator host add <host> [label label ...]
```

Comment:
Regarding adding labels to `host add`:

* In some orchestrators, host / node labels are something more dynamic: you can add and remove them at any time. By making this a parameter of `add host`, we're very much limiting the use case of labels, especially for day-2 operations.
* In DeepSea for example, adding services is internally implemented using host labels. I'm not yet sure how we want to solve this ambiguity.

In any way, this may be a topic that's a bit more complex. We have to find a common understanding of host labels. Would it be an idea to move this into a seperate PR? I've added this to the topics for today's meeting.

cc @jmolmo @tserong

Reply:
@sebastian-philipp @jmolmo the etherpad is still down, but IIRC from Monday's orchestrator meeting the consensus was that we should allow each backend to manage labels how it sees fit, and then rely on the semantics of the interfaces to make the connection between daemon types (e.g. create_osd create_nsf etc...) and labels (or whatever grouping mechanism an orchestrator uses).

Though it does start to raise the question as to why an interface like `add_host` is needed if we have `create_osd,mon,mgr,etc..` anyway?

I will split this out into a separate PR so we can discuss.

Reply:
> Though it does start to raise the question as to why an interface like `add_host` is needed if we have `create_osd,mon,mgr,etc..` anyway?

Yes, this is what I was saying a few weeks ago in one of our meetings. IMO "host management" should be done directly in the orchestrator backend module. That keeps the orchestrator interface more compatible with the very different backends (like containers, and bare-metal).

Reply:
@rjfd: This will imply to "discover" the "orchestrated" cluster and all the services provided when the orchestrator starts...
Agree with you that this simplify interface but forces to have more "intelligence" in the orchestrator modules for all the operations.... but in fact, this is what is intended to do in this kind of  modules.
So :+1:

Reply:
we're going to need host management for `device ls` to let the orchestrator return the c-v inventory.

Reply:
Is it then the case that we should remove labels from the `InventoryFilter` interface? It seems labels are either a part of the orchestrator abstraction, or they should be completely removed.